conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowTypeFunctions.java

```diff
@@ -259,6 +265,18 @@ public Boolean visitUnknown(String unknownType) {
         }
     }
 
+    // TODO(ckozak): Better algorithm to opt into async functionality
```

Comment:
Looking for better ideas to opt into this feature.

My preference is to allow it to be configured per endpoint without requiring us to generate entire apis as either blocking or synchronous. I don't want developers to split a single service API into multiple services for convenience when some endpoints benefit from asynchronous processing.

Reply:
I'm somewhat skeptical this is the right choice, because it complicates the conjure configuration space and the workaround is as simple as `return Futures.immediateFuture(syncReturnValue)`

Reply:
I agree that this is dirty, and I don't think it would be problematic for foundry if we used a global switch (e.g. `--undertowListenableFutures`).

I'm responsible for a project with ~1000 endpoints, the majority of which exist in a single module, ~60 of these endpoints require asynchronous request processing in order to avoid quickly saturating the server worker pool.

Thoughts on this approach?
1. Implement the `--undertowListenableFutures` CLI flag to make every method return ListenableFuture.
2. Implement second `--experimentalUndertowAsyncMarkers` flag which can be enabled to opt into asynchronous processing per-endpint. I will tighten the marker values to accept only an external-type-import annotation to avoid allowing markers to grow beyond existing usage on parameters.

Reply:
ok